lib/proxy/libvirt_network.rb

```diff
@@ -0,0 +1,72 @@
+require 'libvirt'
+
+module Proxy
+  class LibvirtNetwork
+    include Proxy::Log
+    include Proxy::Util
+
+    def initialize(url = nil, network = nil)
+      @connection = ::Libvirt::open(url) if url
+      @network = network if network
+    end
+
+    def dump_xml
+      find_network.xml_desc
+    end
+
+    def dhcp_leases
+      find_network.dhcp_leases
+    end
+
+    def add_dhcp_record(record)
+      nametag = "name=\"#{record.name}\"" if record.name
+      xml = "<host mac=\"#{record.mac}\" ip=\"#{record.ip}\" #{nametag}/>"
+      network_update ::Libvirt::Network::UPDATE_COMMAND_ADD_LAST, ::Libvirt::Network::NETWORK_SECTION_IP_DHCP_HOST, xml
+    rescue ::Libvirt::Error => e
+      logger.error msg = "Error adding DHCP record: #{e}"
+      raise Proxy::DHCP::Error, msg
```

Comment:
This (and other similar methods below) make core class depend on a module. One way to handle this would be to move the methods into subclasses in respective modules.